server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraSpamAssassinContractTest.java

```diff
@@ -19,9 +19,13 @@
 package org.apache.james.jmap.cassandra;
 
 import org.apache.james.jmap.methods.integration.SpamAssassinContract;
-import org.junit.jupiter.api.extension.ExtendWith;
+import org.apache.james.spamassassin.SpamAssassinExtension;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
-@ExtendWith(CassandraJmapExtension.class)
-public class CassandraSpamAssassinContractTest implements SpamAssassinContract {
+class CassandraSpamAssassinContractTest implements SpamAssassinContract {
 
+    @RegisterExtension
+    static SpamAssassinExtension spamAssassin = new SpamAssassinExtension();
+    @RegisterExtension
+    static JamesOnCassandraWithSpamAssassinExtension testExtension = new JamesOnCassandraWithSpamAssassinExtension(spamAssassin);
```

Comment:
anyway, I still have to use 2 `@RegisterExtension` because contract test requires 2 parameters: `GuiceJamesServer` and `SpamAssassin`. While `JamesServerExtension` created from `JamesServerExtensionBuilder` supports only one

Reply:
@trantienduchn you could create a method in the `SpamAssassinContract` to pass the `SpamAssassin` parameter.

That way you could achieve the same functionality with only one registered extension.

Reply:
thanks for your suggestions, rebased to bc268ae